includes/widgets/video.php

```diff
@@ -788,37 +797,42 @@ protected function render() {
 
 		if ( 'hosted' === $settings['video_type'] ) {
 			$video_url = $this->get_hosted_video_url();
+		} else {
+			$embed_params = $this->get_embed_params();
+			$embed_options = $this->get_embed_options();
 		}
 
 		if ( empty( $video_url ) ) {
 			return;
 		}
 
-		if ( 'hosted' === $settings['video_type'] ) {
-			ob_start();
-
-			$this->render_hosted_video();
-
-			$video_html = ob_get_clean();
+		if ( 'youtube' === $settings['video_type'] ) {
+			$video_html = '<div class="elementor-video"></div>';
 		} else {
-			$embed_params = $this->get_embed_params();
+			if ( 'hosted' === $settings['video_type'] ) {
+				$this->add_render_attribute( 'video-wrapper', 'class', 'e-hosted-video' );
 
-			$embed_options = $this->get_embed_options();
+				ob_start();
 
-			$is_static_render_mode = Plugin::$instance->frontend->is_static_render_mode();
-			$post_id = get_queried_object_id();
+				$this->render_hosted_video();
 
-			if ( $is_static_render_mode ) {
-				$video_html = Embed::get_embed_thumbnail_html( $video_url, $post_id );
+				$video_html = ob_get_clean();
 			} else {
-				$video_html = Embed::get_embed_html( $video_url, $embed_params, $embed_options );
+				$is_static_render_mode = Plugin::$instance->frontend->is_static_render_mode();
```

Comment:
The static render should work also for youtube videos.
It's recommended to create a Video_Base class that each provider can extend it.

Reply:
Will be considered as a maintenance task